pennylane/tape/tapes/tape.py

```diff
@@ -168,6 +168,7 @@ class QuantumTape(AnnotatedQueue):
 
     Args:
         name (str): a name given to the quantum tape
+        embed (bool): Whether to embed this tape in a parent tape context.
```

Comment:
nice solution 👍

Reply:
thank you

Reply:
Actually, do you know if this will behave correctly in deeply nested contexts?

Say I had:

```python
with tape(embed=True) as a
  with tape(embed=False) as c:
    with tape(embed=True) as d:
       pass
```

Would `c` embed itself in `b` or `a`?

Reply:
Yep, it doesn't work as I expect:

```python
import pennylane as qml
from pennylane import numpy as np


with qml.tape.QuantumTape(embed=True) as a:
  with qml.tape.QuantumTape(embed=False) as b:
    with qml.tape.QuantumTape(embed=True) as c:
       pass

print(a._ops)
print(b._ops)
```

gives

```
[]
[<QuantumTape: wires=[], params=0>]
```

I guess I would've expected c to be appended to a, as in the `TapeOperationRecorder`. Or would you instead expect the behaviour as is?

Reply:
Aw perfect. No, this is the behavior I wanted/expected. This should allow me to do the arbitrarily nesting

Reply:
Nice. I suggest updating the docstring to make it clear that this is the behaviour

Reply:
I'm worried we might one day feel about `embed` like we do about `do_queue` :laughing:

Reply:
> What's wrong with do_queue?

I think it's more conceptual - in the new queuing system, the idea is _everything_ gets queued regardless, but you can annotate the queue if specific ops should be ignored. As the annotated queue architect, is that the idea @co9olguy?